addon/mixins/infinity-custom.js

```diff
@@ -0,0 +1,85 @@
+import Ember from 'ember';
+import InfinityRoute from 'ember-infinity/mixins/route';
+
+const assign = Ember.assign || Ember.merge;
+
+/**
+ * A custom overlay on ember-infinity that supports loading infinite and paginated relationships
+ * For the most part, the API and semantics are identical to ember infinity, except that the means of configuring the store find method is more flexible
+ *  (supporting relationship queries that do not operate via store methods)
+
+ @class RouteMixin
+ @namespace EmberOSF
+ @module ember-osf/mixins/infinity-custom
+ @extends Ember.Mixin, InfinityRoute
+ */
+const RouteMixin = Ember.Mixin.create(InfinityRoute, {
+    /**
+     * Repurpose an ember-infinity hook to override the method used for queries
+     * @type {function}
+     * @default this.store.query
+     */
+    _storeFindMethod: null,
+
+    /**
+     Use the infinityModel method in the place of `this.store.find('model')` to
+     initialize the Infinity Model for your route.
+
+     @method infinityModel
+     @param {String} modelName The name of the model.
+     @param {Object} options Optional, the perPage and startingPage to load from.
+     @param {Object} boundParams Optional, any route properties to be included as additional params.
+     @return {Ember.RSVP.Promise}
+     */
+    infinityModel(modelName, options, boundParams) {
+        this.set('_infinityModelName', modelName);
+
+        options = options ? assign({}, options) : {};
+        const startingPage = options.startingPage === undefined ? 0 : options.startingPage - 1;
```

Comment:
Yeah. This while method would be better off not needing to exist... it's mostly a cut-and-paste tweak to the parent method. 

The main thing this custom class does is to make the store query method configurable (inserted in the middle of the function), so we can query relationships. (which don't go through `this.store.query`)

In the next iteration, I may take another crack at consolidating this duplicate code.